torch/csrc/autograd/compiled_autograd.h

```diff
@@ -0,0 +1,583 @@
+#pragma once
+#include <c10/core/impl/TorchDispatchModeTLS.h>
+#include <torch/csrc/autograd/engine.h>
+#include <torch/csrc/utils/python_stub.h>
+#include <torch/csrc/utils/torch_dispatch_mode.h>
+#include <iostream>
+#include <typeindex>
+#include <vector>
+
+// see [Note: Compiled Autograd]
+
+namespace torch {
+namespace autograd {
+namespace generated {
+struct TypeAndSize;
+}
+
+struct SizeInput {
```

Comment:
Why not use SymInt ?

Reply:
We need both a dynamic handling (set independently of if the SymInt object is dynamic) and the concrete real value.

Reply:
A little comment explaining this would be nice